library/Faktory/Worker.hs

```diff
@@ -190,11 +190,12 @@ processorLoop f = do
     Left err -> liftIO $ settingsLogError settings $ "Invalid Job: " <> err
     Right Nothing -> liftIO $ threadDelaySeconds $ settingsIdleDelay workerSettings
     Right (Just job) ->
-      processAndAck job
-        `catches` [ Handler $ \(ex :: WorkerHalt) -> throw ex
-                  , Handler $ \(ex :: SomeException) ->
-                    failJob job $ T.pack $ show ex
-                  ]
+      withException
+        (withException
+          (processAndAck job)
+          (\(ex :: WorkerHalt) -> throw ex))
```

Comment:
I don't think you need this inner `withException` anymore.  It's just going to re-throw, but that's something 
`withException` should handle anyway. 

I think the previous behavior is that we only fail if it's not halting (though I'm not sure why we don't fail if the worker is halted).  `catches` will only invoke the first handler that matches. We may want to do something like the following.

```haskell
processAndAck job
  `withException` (\(ex :: SomeException) ->
      case fromException ex of
        Just (_ :: WorkerHalt) ->  pass
        Nothing -> failJob job $ T.pack $ show ex
    )
```

Granted, if we are okay with failing jobs when the worker gets halted you can just remove the inner `withException` and be done with it.

Reply:
oh i didn't realize`catches` only runs the first found handler. I think we still want to catch `WorkerHalt` so we're not breaking anything.